metric/unaggregated/types.go

```diff
@@ -92,17 +93,20 @@ type GaugeWithPolicies struct {
 // which determines which value field is valid. We intentionally do not use value
 // pointers and nil checks to determine which type is valid in order to avoid the GC
 // overhead of marking and sweeping the metrics.
+// NB(xichen): possibly use refcounting to replace explicit ownership tracking.
 type MetricUnion struct {
 	Type          Type
 	ID            metric.ID
 	CounterVal    int64
 	BatchTimerVal []float64
 	GaugeVal      float64
+	OwnsID        bool
```

Comment:
nit: Since we're going to be creating a lot of `MetricUnion`s and will likely want to pool them, perhaps we should rearrange the fields slightly to optimize their size. The current definition has a size of 64 bytes since Go lays out structs in the order of the fields and does not optimize alignment, if we changed the fields to the following we would only use 56 bytes however (https://play.golang.org/p/S4Nlu7o_b1):

```
type MetricUnion struct {
	Type          byte
	OwnsId        bool
  	ID            []byte
  	CounterVal    int64
  	BatchTimerVal []float64
  	GaugeVal      float64
 	TimerValPool  interface{}
}
```

On a side note, it stinks that Go doesn't have true unions, could lead to some big savings here otherwise.